java/cuvs-java/src/main/java22/com/nvidia/cuvs/internal/CagraIndexImpl.java

```diff
@@ -489,6 +587,94 @@ public CuVSResourcesImpl getCuVSResources() {
     return resources;
   }
 
+  /**
+   * Gets the kNN graph from the index.
+   */
+  @Override
+  public int[][] getGraph() throws Throwable {
+    checkNotDestroyed();
+
+    try (var localArena = Arena.ofConfined()) {
+      // Get index size and graph degree
+      MemorySegment sizePtr = localArena.allocate(C_INT);
+      MemorySegment graphDegreePtr = localArena.allocate(C_INT);
+
+      int returnValue = cuvsCagraIndexGetSize(cagraIndexReference.getMemorySegment(), sizePtr);
+      checkCuVSError(returnValue, "cuvsCagraIndexGetSize");
+
+      returnValue =
+          cuvsCagraIndexGetGraphDegree(cagraIndexReference.getMemorySegment(), graphDegreePtr);
+      checkCuVSError(returnValue, "cuvsCagraIndexGetGraphDegree");
+
+      int size = sizePtr.get(C_INT, 0);
+      int graphDegree = graphDegreePtr.get(C_INT, 0);
+
+      // Allocate memory for the graph
+      long graphElements = (long) size * graphDegree;
+      Arena arena = resources.getArena();
+      SequenceLayout graphSequenceLayout = MemoryLayout.sequenceLayout(graphElements, C_INT);
+      MemorySegment graphMemorySegment = arena.allocate(graphSequenceLayout);
+
+      // Allocate device memory for the graph
+      MemorySegment graphD = arena.allocate(C_POINTER);
```

Comment:
Same here, I think `cuvsCagraIndexGetGraph` does this for you; this time even more important, as (I think) this will save a GPU memory copy.